src/libraries/Microsoft.PowerFx.Interpreter/Functions/LibraryMath.cs

```diff
@@ -25,13 +25,18 @@ private interface IAggregator
         {
             void Apply(FormulaValue value);
 
+            FormulaValue GetDefault(IRContext context);
+
+            FormulaValue NoElementValue(IRContext context);
```

Comment:
Do these need to be on the interface? Does somebody ever call these directly or is it always used via GetResult()

We want to keep interfaces as small as possible. (easy to implement).

Reply:
GetDefault isn't called, I'll remove this.
NoElementValue is called in private static FormulaValue RunAggregator(IAggregator agg, IRContext irContext, FormulaValue[] values)

Ideally, these aggregators should be rewritten using generics to avoid so much duplicated code

public class Aggregator<F, T> : IAggregator<T>
    where F : IAggregatorFunction<T>, new()
    where T : new()

public interface IAggregator<T>
{
    void AddValue(T element);
    T Aggregate();
}

public interface IAggregatorFunction<T>
{
    T Aggregate(List<T> elements);
}

public class Sum<T> : IAggregatorFunction<T>
    where T : ISum<T>, new()

...